src/rollup/index.ts

```diff
@@ -363,7 +364,10 @@ export default function rollup (rawInputOptions: GenericConfigObject) {
 				return result;
 			});
```

Comment:
Another issue I observed (the best location for this comment would actually be at line 237) is that implicitly, this feature requires BOTH experimental code-splitting AND input to be an array. That is, if I just provide a single entry point not wrapped in an array, nothing will happen. Ideally I would expect this feature to work the following way:
* `experimentalPreserveModules` implicitly enables `experimentalCodeSplitting`
* If `experimentalPreserveModules` is enabled and `input` is not an array, it is treated as an array of length 1.
This should mostly be possible by changing the check in line 237 and transforming the input if necessary.

Reply:
fixed